### DISCUSSION TITLE
Configure JSON output in more detail to optimize storage?

### DISCUSSION THREAD

**Discussion by Visiting4776** [GoogleChrome/lighthouse#12151] [Q&A]
I haven't found anything about this in the docs - apologies if I missed anything!
We're running lighthouse, via the node module, on a list of websites, weekly and 5 times per website. We save the output as JSON-files to get the full details that aren't available from just saving as CSV. 

Is there any way to reduce the file size of the generated JSON files as they are created? For example, the description for each audit is saved in each JSON file, creating redundancies. Also, the screenshot feature is really awesome, but we would prefer only the final screenshot to be saved.

Presumably this would be done through the `settings` property of a custom configuration file, but I haven't found the option for a lighter JSON output [here](https://github.com/GoogleChrome/lighthouse/blob/575e29b8b6634bfb280bc820efea6795f3dd9017/types/externs.d.ts#L141-L186).

I'm aware of the config option 'onlyCategories', but I'm trying to collect as much information as possible while reducing redundancies. Currently I have my node script strip the descriptions from the saved reports after the fact, I'm just wondering if there is a better option. Thank you for your attention and this tool, it's really useful!

**Comment by patrickhulce** [comment]
There are no options or settings that automatically reduce the volume of the JSON beyond the filters that you've already discovered (`onlyCategories`/`onlyAudits`/`skipAudits`).

What you can trim is highly dependent on the use case. There is plenty of data you might not be using that will be more effective to remove than just the string descriptions. Most mass collections choose to [strip the embedded screenshots](https://github.com/HTTPArchive/bigquery/blob/17a1bcd43e988f001405810be83ed71f3428db75/dataflow/java/src/main/java/com/httparchive/dataflow/BigQueryImport.java#L279-L286), full page screenshot, debug timings, i18n data, and just compress the rest.

You'll probably want to just go through the [LHR definition](https://github.com/GoogleChrome/lighthouse/blob/0daa1c3b02c6ac6bad6049b89afcd8ecb68e5224/types/lhr.d.ts#L27-L59) and [audit list](https://github.com/GoogleChrome/lighthouse/blob/0daa1c3b02c6ac6bad6049b89afcd8ecb68e5224/lighthouse-core/config/default-config.js#L184-L337) to see what you don't care about.

**Reply by ufukbay** [reply]
Thank you so much for the link to the audit list!
